src/controller/ViewReportsScreenController.java

```diff
@@ -34,7 +43,12 @@ public void registerMainApp(MainFXApplication main) {
      * Updates the reports list with stringified reports.
      */
     public void setReportsList() {
-        Stream<Report> reports = main.getReportManager().getSourceReports();
+        Stream<? extends Report> reports;
+        if (viewPurityReports) {
+            reports = main.getReportManager().getPurityReports();
```

Comment:
This is a violation of the 'tell don't ask' principle in OOP. Instead of asking `main` for the report manager and then the purity reports list from the report manager, we should just pass a reference to the reports we want to display into  the controller. That lets us easily reuse this class for any subclass of report.